### ISSUE TITLE
Install mitmproxy binary if possible

### ISSUE THREAD

**Issue by fabricereix** [Orange-OpenSource/hurl#1414]
Currently mitmproxy (used for integration testing) is installed from a standard `pip install`.
This triggers a build (Rust) that currently failed on alpine.
```
...
         error: could not compile `boringtun` due to 15 previous errors
        warning: build failed, waiting for other jobs to finish...
        💥 maturin failed
          Caused by: Failed to build a native library through cargo
          Caused by: Cargo build finished with "exit status: 101": `"cargo" "rustc" "--release" "--message-format" "json" "--lib"`
        📦 Including license file "LICENSE"
        🔗 Found pyo3 bindings with abi3 support for Python ≥ 3.7
        🐍 Not using a specific python interpreter
        error[E0433]: failed to resolve: could not find `StaticSecret` in `x25519_dalek`
           --> /root/.cargo/registry/src/github.com-1ecc6299db9ec823/boringtun-0.5.2/src/device/api.rs:226:62
            |
        226 | ...                   device.set_key(x25519_dalek::StaticSecret::from(key_bytes.0))
            |                                                    ^^^^^^^^^^^^ could not find `StaticSecret` in `x25519_dalek`
        
        
        error[E0412]: cannot find type `StaticSecret` in crate `x25519_dalek`
           --> /root/.cargo/registry/src/github.com-1ecc6299db9ec823/boringtun-0.5.2/src/device/mod.rs:118:37
            |
```
For external tools, it would probably make sense to get pre-built binaries to avoid breaking our build itself.

**Comment by jcamiel** [comment]
Raised a discussion on mitmproxy repo => https://github.com/mitmproxy/mitmproxy/discussions/6045

**Comment by jcamiel** [comment]
Closed by #1431 (use squid instead of mitmproxy in CI)
